stripped/colabs/Agregacja_stripped.py

```diff
@@ -0,0 +1,586 @@
+from sklearn.model_selection import train_test_split
+from sklearn.feature_selection import mutual_info_classif
+from typing import Optional, Tuple, List
+from imblearn.over_sampling import RandomOverSampler, SMOTE
+from bokeh.layouts import row
+
+from google.colab import drive
+drive.mount('/content/drive')
+
+!git clone https://github.com/PiotrMaciejKowalski/BigData2024Project.git
+
+!chmod 755 /content/BigData2024Project/src/setup.sh
+!/content/BigData2024Project/src/setup.sh
+
+import sys
+sys.path.append('/content/BigData2024Project/src')
+
+from start_spark import initialize_spark
+initialize_spark()
+
+from big_mess.loaders import default_loader
+from pyspark.sql import SparkSession, DataFrame as SparkDataFrame, functions as F
+
+spark = SparkSession.builder\
+        .master("local")\
+        .appName("Colab")\
+        .config('spark.ui.port', '4050')\
+        .getOrCreate()
+
+%%time
+nasa = default_loader(spark)
+# nasa.show(5)
+
+nasa.createOrReplaceTempView("nasa")
+
+from big_mess.loaders import load_anotated
+nasa_anotated = load_anotated(spark)
+
+nasa_anotated.createOrReplaceTempView("nasa_anotated")
+
+def nasa_agg_year_with_avg(sdf: SparkDataFrame, year: int) -> SparkDataFrame:
+    """
+    Funkcja sluzaca do ograniczenia zbioru NASA do jednego roku,a następnie wykonująca agregacje roczna:
+    suma miesieczna -> suma roczna (Rainf, Evap)
+    srednia miesieczna -> srednia roczna (pozostale zmienne)
+    """
+    data = sdf.filter(f'Year = {year}')
+    data.createOrReplaceTempView("data")
+    data_group_by = spark.sql("""
+          SELECT
+          lon, lat,
+          SUM(Rainf) AS Rainf,
+          SUM(Evap) AS Evap,
+          AVG(AvgSurfT) AS AvgSurfT,
+          AVG(Albedo) AS Albedo,
+          AVG(SoilT_40_100cm) AS SoilT_40_100cm,
+          AVG(GVEG) AS GVEG,
+          AVG(PotEvap)As PotEvap,
+          AVG(RootMoist) AS RootMoist,
+          AVG(SoilM_100_200cm) AS SoilM_100_200cm
+          FROM data
+          GROUP BY lon, lat
+          ORDER BY lon, lat
+          """)
+    return data_group_by
+
+def nasa_agg_year_with_median(sdf: SparkDataFrame, year: int) -> SparkDataFrame:
+    """
+    Funkcja sluzaca do ograniczenia zbioru NASA do jednego roku,a następnie wykonująca agregacje roczna:
+    suma miesieczna -> suma roczna (Rainf, Evap)
+    srednia miesieczna -> mediana (pozostale zmienne)
+    """
+    data = sdf.filter(f'Year = {year}')
+    data.createOrReplaceTempView("data")
+    data_group_by = spark.sql("""
+          SELECT
+          lon, lat,
+          SUM(Rainf) AS Rainf,
+          SUM(Evap) AS Evap,
+          MEDIAN(AvgSurfT) AS AvgSurfT,
+          MEDIAN(Albedo) AS Albedo,
+          MEDIAN(SoilT_40_100cm) AS SoilT_40_100cm,
+          MEDIAN(GVEG) AS GVEG,
+          MEDIAN(PotEvap)As PotEvap,
+          MEDIAN(RootMoist) AS RootMoist,
+          MEDIAN(SoilM_100_200cm) AS SoilM_100_200cm
+          FROM data
+          GROUP BY lon, lat
+          ORDER BY lon, lat
```

Comment:
```
          ORDER BY lon, lat
```
a to do czego?

Reply:
W zasadzie dla agregacji z medianami nie było potrzebne, odruchowo przekopiowałam. Natomiast w agregacji z AVG rozrzucało mi kolejność wierszy (zbiór nasa dla każdego miesiąca jest domyślnie posortowany według lon, lat), a to się przyczyniało do tego że zamiast otrzymać taki wykres:
![image](https://github.com/PiotrMaciejKowalski/BigData2024Project/assets/148125699/c5a65982-9437-4eb3-9f97-79d0e97cf271)
miałam taki:
![image](https://github.com/PiotrMaciejKowalski/BigData2024Project/assets/148125699/ecd27817-5d7e-4c06-9cee-b947d194dfe9)
i mi się nie podobało.